lib/chat_app_web/live/single_room_live.ex

```diff
@@ -91,4 +110,21 @@ defmodule ChatAppWeb.SingleRoomLive do
         end)
      )}
   end
+
+  def handle_info({:users_typing, username, status}, socket) do
+    cond do
+      socket.assigns.username == username -> {:noreply, socket}
+      socket.assigns.username != username ->
+        current_status = MapSet.member?(socket.assigns.users_typing, username)
+        animation_condition = current_status != status or current_status == status and current_status
+        users_typing = socket.assigns.users_typing
+        updated_typing_statuses = if not status, do: MapSet.delete(users_typing, username), else: MapSet.put(users_typing, username)
+        updated_socket = assign(socket, users_typing: updated_typing_statuses)
```

Comment:
Please avoid using `cond` where it isn't necessary. a `case` will do just fine :)